Detectors/Upgrades/ALICE3/FCT/simulation/src/Detector.cxx

```diff
@@ -0,0 +1,592 @@
+    }
+  }
+}
+
+//_________________________________________________________________________________________________
+void Detector::buildBasicFCT(const FCTBaseParam& param)
+{
+  // Build a basic parametrized FCT detector with nLayers equally spaced between z_first and z_first+z_length
+  // Covering pseudo rapidity [etaIn,etaOut]. Silicon thinkness computed to match layer x/X0
+
+  LOG(info) << "Building FCT Detector: Conical Telescope";
+
+  auto z_first = param.z0;
+  auto z_length = param.zLength;
+  auto etaIn = param.etaIn;
+  auto etaOut = param.etaOut;
+  auto Layerx2X0 = param.Layerx2X0;
+  mNumberOfLayers = param.nLayers;
+  mLayerID.clear();
+
+  Int_t type = 0; // Disk
+
+  for (Int_t layerNumber = 0; layerNumber < mNumberOfLayers; layerNumber++) {
+    std::string layerName = GeometryTGeo::getFCTLayerPattern() + std::to_string(layerNumber); // + mNumberOfLayers * direction);
+    mLayerName.push_back(layerName);
+
+    // Adds evenly spaced layers
+    Float_t layerZ = z_first + (layerNumber * z_length / (mNumberOfLayers - 1)) * std::copysign(1, z_first);
+    Float_t rIn = std::abs(layerZ * std::tan(2.f * std::atan(std::exp(-etaIn))));
+    Float_t rOut = std::abs(layerZ * std::tan(2.f * std::atan(std::exp(-etaOut))));
+    auto& thisLayer = mBackwardLayers.emplace_back(0, layerNumber, layerName, layerZ, rIn, rOut, Layerx2X0, type);
+  }
+}
+
+//_________________________________________________________________________________________________
+void Detector::buildFCTV1()
+{
+  // Build default FCT detector
+
+  LOG(info) << "Building FCT Detector: V1";
+
+  mNumberOfLayers = 9;
+  Float_t layersx2X0 = 1.e-2;
+
+  std::vector<std::array<Float_t, 4>> layersConfig{
+    {-442.0, 6.0, 44.1, layersx2X0}, // {z_layer, r_in, r_out, Layerx2X0}
+    {-444.0, 6.0, 44.3, layersx2X0},
+    {-446.0, 6.0, 44.5, layersx2X0},
+    {-448.0, 6.0, 44.7, layersx2X0},
+    {-450.0, 6.1, 44.9, layersx2X0},
+    {-460.0, 6.2, 45.9, layersx2X0},
+    {-470.0, 6.3, 46.9, layersx2X0},
+    {-480.0, 6.5, 47.9, layersx2X0},
+    {-490.0, 6.6, 48.9, layersx2X0}};
+
+  mLayerID.clear();
+  mLayerName.clear();
+  mBackwardLayers.clear();
+
+  Int_t type = 0; // Disk
+
+  for (int layerNumber = 0; layerNumber < mNumberOfLayers; layerNumber++) {
+    std::string directionName = std::to_string(0);
+    std::string layerName = GeometryTGeo::getFCTLayerPattern() + directionName + std::string("_") + std::to_string(layerNumber);
+    mLayerName.push_back(layerName);
+    Float_t z = layersConfig[layerNumber][0];
+
+    Float_t rIn = layersConfig[layerNumber][1];
+    Float_t rOut = layersConfig[layerNumber][2];
+    Float_t x0 = layersConfig[layerNumber][3];
+
+    LOG(info) << "Adding Layer " << layerName << " at z = " << z;
+    // Add layers
+    auto& thisLayer = mBackwardLayers.emplace_back(0, layerNumber, layerName, z, rIn, rOut, x0, type);
```

Comment:
you are not using `thisLayer`.
Only `mBackwardLayers.emplace_back(0, layerNumber, layerName, layerZ, rIn, rOut, Layerx2X0, type);` will do

Reply:
Done